### PR TITLE
feat: add sample test for mobile

### PR DESCRIPTION
**Pull request by helenren**

### CODE REVIEW

File: package.json
Comment (helenren) [inline package.json:38]: once the salesforce-pageobjects includes the mobile pos, will remove this one

**Comment by helenren** [comment]
@olivier-martin-sf this change is ready to review, pls have a look. Thanks.

File: force-app/test/mobile/login.spec.ts
Comment (olivier-martin-sf) [inline force-app/test/mobile/login.spec.ts:10]: nit: IIRC you can set pass the config specifier directly to `setLoaderConfig` but the way you did is also totally fine

File: force-app/test/mobile/login.spec.ts
Comment (olivier-martin-sf) [inline force-app/test/mobile/login.spec.ts:19]: nit: I would rename the `isPresent` variable to something more meaningful

**Comment by olivier-martin-sf** [comment]
@mleonardsfdc if you have some time, could you please do a review of the README in this PR?

File: wdio.conf.ios.js
Comment (olivier-martin-sf) [inline wdio.conf.ios.js:10]: I would suggest to declare a base config that export an object with all shared properties with IOS and Android and then merge this config with the specific options for each to avoid having duplicating config settings. I don't know if WDIO has some kind of `extends` or `overrides` built-in but a plain JS object merge would just work:

```js
{
   ...baseMobileConfig,
   iosMobileConfig
}
```

File: package.json
Comment (olivier-martin-sf) [reply package.json:38]: Is `salesforceapp-pageobjects` published externally though? Is it on npm or only internally in Nexus?

File: README.md
Comment (olivier-martin-sf) [inline README.md:288]: nit: typo it's node.js

Comment (olivier-martin-sf) [reply README.md:288]: suggestion: I would add the link for both project homepage

File: README.md
Comment (olivier-martin-sf) [inline README.md:289]: nit: typo `configure` -> `configuration`

File: README.md
Comment (olivier-martin-sf) [inline README.md:290]: suggestion: enclose the config filename in backticks

File: README.md
Comment (olivier-martin-sf) [inline README.md:298]: suggestion: enclose the config filename in backticks

**Review by olivier-martin-sf — CHANGES_REQUESTED** [review]
Thx Qingchun, it looks good to me overall the only concern I have merging this is about the `salesforceapp-pageobjects` package that's not available on npm which would cause installation failure for the customers

File: package.json
Comment (helenren) [reply package.json:40]: is there any reason, that we cannot use ts here? @olivier-martin-sf

File: wdio.conf.ios.js
Comment (helenren) [reply wdio.conf.ios.js:3]: updated

File: package.json
Comment (helenren) [reply package.json:38]: yes, this is internal only. Will not merge this until salesforce-pageobjects includes the mobile pos. Just use this one here to show how to demonstrate how to use it in test.

Comment (olivier-martin-sf) [reply package.json:38]: Sounds good! thank you for the explanations! I will resolve this conversation

File: package.json
Comment (olivier-martin-sf) [reply package.json]: It's just to make things as simple as possible and put the emphasis on the page objects library and how to use it. The repo is small and quite straightforward so JS is a good candidate actually. Most of the TS benefits (`type checking`, inference, intellisense...) can also be applied to JS files (that's why we have a `jsconfig.json` file). 

On a side note, TS has actually some perf issues in type checking (I don't think that would be noticeable for a repo this size) and their support for some features around ES Modules is actually still very much WIP. I don't want to have to redirect customers to TS issues instead of focusing on what this repo is about.

Hopefully that gives some reasonable explanations

File: README.md
Comment (olivier-martin-sf) [inline README.md:288]: nit: does it matter that they are installed in the user's home folder? I can tell it doesn't for node but I have no idea for appium

File: README.md
Comment (olivier-martin-sf) [inline README.md:311]: nit: I would hoist that before the test commands to make sure people read the info in the order they have to execute them

File: force-app/test/mobile/login.spec.js
Comment (olivier-martin-sf) [inline force-app/test/mobile/login.spec.js:1]: please add Salesforce copyright headers

File: wdio.conf.android.js
Comment (olivier-martin-sf) [inline wdio.conf.android.js:1]: please add Salesforce copyright headers

File: wdio.conf.ios.js
Comment (olivier-martin-sf) [inline wdio.conf.ios.js:1]: please add Salesforce copyright headers

File: wdio.conf.shared.js
Comment (olivier-martin-sf) [inline wdio.conf.shared.js:1]: please add Salesforce copyright headers

File: package.json
Comment (olivier-martin-sf) [inline package.json:27]: question: why do we need this dependency for?

File: package.json
Comment (olivier-martin-sf) [inline package.json:23]: do we need to actually start the appium server ourselves when it's the purpose of this actual package?

**Review by olivier-martin-sf — CHANGES_REQUESTED** [review]
Left some comments (questions) but requesting changes to includes salesforce header

File: package.json
Comment (helenren) [reply package.json:27]: it is the service needed local run

File: package.json
Comment (helenren) [reply package.json:23]: for using local test device, yes we need a appium server locally.

File: wdio.conf.android.js
Comment (olivier-martin-sf) [inline wdio.conf.android.js:2]: please update the year to 2022

File: wdio.conf.ios.js
Comment (olivier-martin-sf) [inline wdio.conf.ios.js:2]: same comment about the year

File: package.json
Comment (olivier-martin-sf) [reply package.json:23]: I mean do you really need to start the server locally? I though this service is spawning an instance of the appium server automatically

File: wdio.conf.android.js
Comment (helenren) [reply wdio.conf.android.js:2]: still using 2021 in wdio.conf.js, that's I put 2021 here as well. It should be 2022 since adding it in 2022?

File: package.json
Comment (helenren) [reply package.json:23]: yes, we need this to launch an appium server via this service in local when running test. Otherwise, need to manually launch one before test.

File: README.md
Comment (mleonardsfdc) [inline README.md:288]:
```suggestion
- Make sure that [Appium](https://github.com/appium/appium#readme) and [node.js](https://nodejs.org/en/) are installed on your machine.
```

File: README.md
Comment (mleonardsfdc) [inline README.md:290]:
```suggestion
For an iOS test, update `wdio.conf.ios.js` file to configure the test device name, iOS version, and the full path for the test application:
```

Comment (mleonardsfdc) [reply README.md:290]: I'm not clear which of the properties listed above correspond to the example below so it would be good to list them in the same order in the text intro and the js block below. something like:
For an iOS test, update `wdio.conf.ios.js` file to configure the test device name (`appium:deviceName`), the full path for the test application (`appium:app`), and the iOS version (`appium:platformVersion` TODO: not sure if I'm listing the correct prop here so please confirm):

File: README.md
Comment (mleonardsfdc) [inline README.md:298]:
```suggestion
For an Android test, update the `wdio.conf.android.js` file to configure the application bundleid, the full path for the test application, and the application's initial activity:
```

Comment (mleonardsfdc) [reply README.md:298]: Is application bundleid = appium:deviceName or appium:appPackage? It would be helpful to list the corresponding prop in parenthesis in the intro text like in the previous example.

We should also mention the appium:deviceName in the intro text.

I'm not sure what the "application's initial activity" means. How is it used in the test?

File: README.md
Comment (mleonardsfdc) [inline README.md:307]:
```suggestion
- Download the [debug build](https://developer.salesforce.com/tools/mobile-debugging) for the Salesforce iOS and Android mobile apps.
```

File: README.md
Comment (mleonardsfdc) [inline README.md:308]:
```suggestion
- Commands to execute a test: 
```

File: README.md
Comment (mleonardsfdc) [inline README.md:311]:
```suggestion
- For a test on Android, make sure to start an emulator before the test run. Otherwise, you will get an error like this: "Error: Failed to create session.
```

**Review by mleonardsfdc — CHANGES_REQUESTED** [review]
I made some copy edits in suggested changes

**Comment by helenren** [comment]
@mleonardsfdc has accepted all your comments.  Please have a check. Thanks.

File: README.md
Comment (mleonardsfdc) [inline README.md:290]: Not sure if I'm matching the correct prop name in the intro to the example but will try below. Please review and then update the intro.
For an iOS test, update `wdio.conf.ios.js` file to configure the test device name (`appium:deviceName`), iOS version (`appium:platformVersion`), and the full path for the test application (`appium:app`):

File: README.md
Comment (mleonardsfdc) [inline README.md:298]: Not sure if I'm matching the correct prop name in the intro to the example but will try below. Please review and then update the intro.
For an Android test, update the `wdio.conf.android.js` file to configure the test device name (`appium:deviceName`), application bundleid (`appium:appPackage` TODO: not sure if I'm matching the correct prop here so please confirm), the full path for the test application (`appium:app`), and the application's initial activity (`appium:appActivity`):

Also, I'm not clear on usage of the application's initial activity so please explain its usage in the test

**Review by mleonardsfdc — CHANGES_REQUESTED** [review]
I added a few comments that maybe weren't clear in my previous review. Please review and reach out if they are not clear.

**Review by mleonardsfdc — APPROVED** [review]

**Review by olivier-martin-sf — APPROVED** [review]
LGTM after al lthe fixes

**Pull request merged**
